zerver/models.py

```diff
@@ -880,9 +882,11 @@ def upload_quota_bytes(self) -> Optional[int]:
         # it as gibibytes (GiB) to be a bit more generous in case of confusion.
         return self.upload_quota_gb << 30
 
+    # `realm` instead of `self` here to make sure the parameters of the cache key
+    # function matches the original method.
     @cache_with_key(get_realm_used_upload_space_cache_key, timeout=3600 * 24 * 7)
-    def currently_used_upload_space_bytes(self) -> int:
-        used_space = Attachment.objects.filter(realm=self).aggregate(Sum("size"))["size__sum"]
+    def currently_used_upload_space_bytes(realm) -> int:
+        used_space = Attachment.objects.filter(realm=realm).aggregate(Sum("size"))["size__sum"]
```

Comment:
Can you confirm that the way `self` works as a keyword is that the current instance will still be received as the first argument even though it is not called `self`?

Reply:
Replying to https://github.com/zulip/zulip/pull/21782#issuecomment-1099461238:
> `self` is a conventional name for the first argument of methods. It does
not affect how the instance will be passed.

I guess the Python 3 tutorial confirms that:

```
Often, the first argument of a method is called self. This is nothing more than a convention: the name self has absolutely no special meaning to Python. Note, however, that by not following the convention your code may be less readable to other Python programmers, and it is also conceivable that a class browser program might be written that relies upon such a convention.
```

I guess the crux of the issue is that a ParamSpec includes the name of positional arguments, since you can pass a positional parameter by name. OK.

Reply:
It is technically a `positional-or-keyword` argument. Actually, I don't think [PEP 612](https://peps.python.org/pep-0612/#paramspec-variables) mentions that there should be an error in this case. It is likely a false positive of mypy. But the workaround here isn't that harmful. I opened an [issue](https://github.com/python/mypy/issues/12591) to see if we can get a fix.